File: translation/tools/buildTemplate.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void build(char *in, FILE *out) {
    char *s = in;
    char *e;
    int list = 0;

    if ( *in == '"' )
        list = 1;

    if (list) {
        e = strstr(in, "\":");
        e[2] = '\0';
        fprintf(out,"%-120s\"\",\n",in);
    } else {
        e = strstr(in, "\",");
        if ( e == NULL ) {
            fprintf(stderr, "input: <%s>", in);
            exit(0);
        }
        e[2] = '\0';
        fprintf(out,"%-120s\"\"],\n",in);
    }

}
    
void main(int argc, char **argv) {
    FILE *in = NULL;
    FILE *out = NULL;
    char line[1024];
    char *s, *t;
    
    if ( argc < 3 ) {
        fprintf(stderr, "Syntax: buildTemplate <input file> <output file>\n");
        exit(1);
    }
    
    if ((in = fopen(argv[1], "r")) == NULL) {
        fprintf(stderr,"can't open %s\n", argv[1]);
        exit(1);
    }

    if ((out = fopen(argv[2], "w")) == NULL) {
        fprintf(stderr,"can't open %s\n", argv[2]);
        exit(1);
    }
    
    // read line by line;
    while((s = fgets(line, sizeof(line),in))) {
        switch (*s) {
            case '"':
            case '[':
                build(s, out);
            break;
            default:
                fprintf(out,"%s",s);
        }
    }
    
    exit(0);
}

File: translation/tools/buildTemplate.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func build(input string, output *os.File) {
	list := strings.HasPrefix(input, "\"")

	if list {
		index := strings.Index(input, "\":")
		if index == -1 {
			fmt.Fprintf(os.Stderr, "Invalid input: %s\n", input)
			os.Exit(1)
		}
		formatted := fmt.Sprintf("%-120s\"\",\n", input[:index+2])
		output.WriteString(formatted)
	} else {
		index := strings.Index(input, "\",")
		if index == -1 {
			fmt.Fprintf(os.Stderr, "Invalid input: %s\n", input)
			os.Exit(1)
		}
		formatted := fmt.Sprintf("%-120s\"\"],\n", input[:index+2])
		output.WriteString(formatted)
	}
}

func main() {
	if len(os.Args) < 3 {
		fmt.Fprintln(os.Stderr, "Syntax: buildTemplate <input file> <output file>")
		os.Exit(1)
	}

	inputFile, err := os.Open(os.Args[1])
	if err != nil {
		fmt.Fprintf(os.Stderr, "Can't open %s: %v\n", os.Args[1], err)
		os.Exit(1)
	}
	defer inputFile.Close()

	outputFile, err := os.Create(os.Args[2])
	if err != nil {
		fmt.Fprintf(os.Stderr, "Can't open %s: %v\n", os.Args[2], err)
		os.Exit(1)
	}
	defer outputFile.Close()

	scanner := bufio.NewScanner(inputFile)
	for scanner.Scan() {
		line := scanner.Text()
		build(line, outputFile)
	}

	if err := scanner.Err(); err != nil {
		fmt.Fprintf(os.Stderr, "Error reading input file: %v\n", err)
		os.Exit(1)
	}
}
